src/processes.c

```diff
@@ -1529,8 +1529,14 @@ static char *ps_get_owner(pid_t pid)
         uid = atoi (line + 5);
         getpwuid_r (uid, &passwd, passwd_buffer, sizeof(passwd_buffer),
                 &passwd_result);
-        if (passwd_result)
+        if (passwd_result) {
             result = sstrdup (passwd_result->pw_name);
+        } else {
+	    // Send the numeric uid if name is not available.
+            char uid_str[6];  // 5 digits plus string terminator.
```

Comment:
(a) The buffer was perfectly fine — because `snprintf` was used, the value might be truncated, but would never overflow.
(b) `strtoul` produces a 32-bit value, and the `"%lu"` format assumes that as well, so `%lu` will never result in more than 10 decimal digits.
(c) If you were worried that `uid_t` would be larger than 32 bits in some cases, you could have used something like `sizeof(uid_t)/3+1` as the static size of the array.